gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -425,6 +434,47 @@ public static List<Service> parseService(
         Transport.GRPC);
   }
 
+  private static Optional<List<String>> getInclusionMethodListFromServiceYaml(
+      Optional<com.google.api.Service> serviceYamlProtoOpt, String protoPackage) {
```

Comment:
It is usually preferred to not passing around `Optional` if we can check it before calling the method.

Reply:
I want to follow-up on this from our discussion earlier and get @blakeli0 your opinion on this. (FYI, I've combined this method into `shouldIncludeMethodInGeneration()` )
For now, I am slightly inclined to keep this `Optional<com.google.api.Service> serviceYamlProtoOpt` as input for 2 reasons: 
- The pattern of passing in this `Optional` is repeated in this `Parser` class, a larger scale refactor is needed to remove these.
- Because this service yaml is an optional config file that only some libraries have, it sort of make sense to have this Optional representation. And  I find the current way can better encapsulate this piece of logic: `shouldIncludeMethodInGeneration()` has all the information to decide if this method should or not be included. I could check `serviceYamlProtoOpt` before calling the method, but the 2 occurrences would repeat the logic in a reverse way, and it might be less readable for debugging in the future. 

Does this makes sense to you? WDYT?